backend/app/Http/Controllers/StudentController.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Student;
+use Illuminate\Http\Request;
+
+class StudentController extends Controller
+{
+  // register a new user
+  public function store(Request $request) {
+    $fields = $request->validate([
+      'name' => 'required|string',
+      'username' => 'required|string',
+      'email' => 'required|string|unique:students,email',
+      'thumbnail' => 'required|image',
+      'password' => 'required|string|confirmed'
+    ]);
+
+    $fields['thumbnail'] = request()->file('thumbnail')->store('thumbnails');
+
+    $student = Student::create([
+      'name' => $fields['name'],
+      'username' => $fields['username'],
+      'email' => $fields['email'],
+      'thumbnail' => $fields['thumbnail'],
+      'password' => bcrypt($fields['password'])
+    ]);
+
+    $token = $student->createToken('myapptoken')->plainTextToken;
+
+    $response = [
+      'student' => $student,
+      'token' => $token
+    ];
```

Comment:
Will the token be used for something?

Reply:
Hello Sir, I intended to use the token, so that I can check if the
registration is successful, once there is a token it will be used as a
condition in if else so that I can display a successful alert or an error
alert.

Reply:
@kentlouise 
ahh okay i understand for this case we can use the return status of the request instead of the token so for this one its okay no need for the token. if status 200 in the frontend then request its successful.

Additional information:
You can set set status code to return with your json content in the 2nd parameter of the function. But by default its status 200
```
// will return json with 200 status
return request()->json('Successfully created.');

// you can change the status of its return like so (example 500 internal server error)
return request()->json('Something went wrong.', 500);

Reply:
Okay got it Sir. One moment.